gslib/wildcard_iterator.py

```diff
@@ -649,8 +651,12 @@ def __iter__(self, bucket_listing_fields=None):
     for filepath in filepaths:
       expanded_url = StorageUrlFromString(filepath)
       try:
-        if self.exclude_pattern is not None:
-          if self.exclude_pattern.match(str(expanded_url)):
+        if self.exclude_tuple is not None:
+          (base_url_str, exclude_pattern) = self.exclude_tuple
+          str_to_check = filepath[len(base_url_str):]
```

Comment:
It seems like base_url_str is a StorageUrl.url_string, which means it will have the scheme prefix i.e `file://`. The `filepath` here is probably just the output of glob, so will be without any scheme, and hence the str_to_check will be incorrect. You might want to use the `expanded_url` instead.

I'm guessing this was probably why the `.*txt$|ok` was working before.

Reply:
Good catch!

Reply:
I remember this was copying the previous implementation very closely so I'll want to test this a bunch before I make further changes.